storage/series/buffer.go

```diff
@@ -592,39 +632,33 @@ func (b *dbBufferBucket) resetBootstrapped() {
 	b.bootstrapped = nil
 }
 
-type discardMergedResult struct {
-	block  block.DatabaseBlock
-	merges int
+func (b *dbBufferBucket) needsMerge() bool {
+	return b.canRead() && !(b.hasJustSingleEncoder() || b.hasJustSingleBootstrappedBlock())
 }
 
-func (b *dbBufferBucket) discardMerged() discardMergedResult {
-	defer func() {
-		b.resetEncoders()
-		b.resetBootstrapped()
-		b.empty = true
-	}()
-
-	if len(b.encoders) == 1 && len(b.bootstrapped) == 0 {
-		// Already merged as a single encoder
-		encoder := b.encoders[0].encoder
-		newBlock := b.opts.DatabaseBlockOptions().DatabaseBlockPool().Get()
-		newBlock.Reset(b.start, encoder.Discard())
-		b.encoders = b.encoders[:0]
-		return discardMergedResult{newBlock, 0}
-	}
+func (b *dbBufferBucket) hasJustSingleEncoder() bool {
+	return len(b.encoders) == 1 && len(b.bootstrapped) == 0
+}
 
-	if (len(b.encoders) == 0 ||
+func (b *dbBufferBucket) hasJustSingleBootstrappedBlock() bool {
+	encodersEmpty := len(b.encoders) == 0 ||
```

Comment:
do the other bits of the code guarantee we won't have multiple empty encoders?

Reply:
Yeah so for a bucket that returns `true from `canRead()` its guarenteed that there's data that is written.